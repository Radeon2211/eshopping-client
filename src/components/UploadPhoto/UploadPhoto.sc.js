import styled from 'styled-components';

export const Wrapper = styled.div`
  margin-bottom: ${({ theme }) => theme.spacings.level3};

  & .error {
    margin-top: ${({ theme }) => theme.spacings.level2};
  }

  & .input {
    display: none;
  }

  & .label {
    cursor: pointer;
    display: inline-block;

    & > button {
      pointer-events: none;
      transition: all ${({ theme }) => theme.durations.level1}s;
    }

    &:hover > button {
      background-color: ${({ theme }) => theme.colors.blueLight};
      border-color: ${({ theme }) => theme.colors.blueLight};
    }
  }
`;

export const Preview = styled.div`
  font-size: ${({ theme }) => theme.fontSizes.level1};
  margin-top: ${({ theme }) => theme.spacings.level2};

  & .file-data:not(:last-child) {
    margin-right: ${({ theme }) => theme.spacings.level3};
  }
`;
