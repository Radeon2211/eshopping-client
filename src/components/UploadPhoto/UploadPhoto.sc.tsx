import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.div`
  margin-bottom: ${({ theme }) => theme.spacings.level3};

  & .input {
    display: none;
  }

  & .label {
    cursor: pointer;
    display: inline-block;

    & > button {
      pointer-events: none;
      transition: all ${({ theme }) => theme.durations.level1}s;
    }

    &:hover > button {
      background-color: ${({ theme }) => theme.colors.blueLight};
      border-color: ${({ theme }) => theme.colors.blueLight};
    }
  }
`;
