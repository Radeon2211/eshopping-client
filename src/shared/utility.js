export const updateObject = (oldObject, updatedProps) => ({
  ...oldObject,
  ...updatedProps,
});

export const getErrorMessage = (error) => {
  let errorMessage = 'Something went wrong';
  if (Object.values(error.response.data?.errors).length > 0) {
    errorMessage = Object.values(error.response.data.errors)[0].properties.message;
  }
  return errorMessage;
};