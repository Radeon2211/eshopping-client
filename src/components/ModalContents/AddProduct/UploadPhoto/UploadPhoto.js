/* eslint-disable react/jsx-one-expression-per-line */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import * as SC from './UploadPhoto.sc';
import { isValidFileType, calculateFileSize } from '../../../../shared/utility';
import Button from '../../../UI/Button/Button';

const PRODUCT_PHOTO_MAX_SIZE = 6291456;
const PRODUCT_PHOTO_MAX_SIZE_STRING = calculateFileSize(PRODUCT_PHOTO_MAX_SIZE);

const UploadPhoto = (props) => {
  const { setFieldValue } = props;

  const [photo, setPhoto] = useState(null);
  const [photoName, setPhotoName] = useState('');
  const [photoSize, setPhotoSize] = useState('');
  const [error, setError] = useState('');

  const resetState = () => {
    setPhoto(null);
    setFieldValue('photo', null);
    setPhotoName('');
    setPhotoSize('');
    setError('');
  };

  const inputChangedHandler = async (e) => {
    const { files } = e.target;
    if (!files.length > 0) {
      resetState();
      return;
    }

    const file = files[0];
    let fileName = file.name;
    if (fileName.length > 30) {
      fileName = `${fileName.slice(0, 25)}...${file.type.split('/')[1]}`;
    }
    setPhotoName(fileName);
    setPhotoSize(calculateFileSize(file.size));

    if (!isValidFileType(file.type)) {
      setPhoto(null);
      setFieldValue('photo', null);
      setError('File extension is not valid (JPG and PNG only)');
      return;
    }
    setError(null);

    if (file.size > PRODUCT_PHOTO_MAX_SIZE) {
      setError(`Maximum available size is ${PRODUCT_PHOTO_MAX_SIZE_STRING}`);
    }

    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onloadend = () => {
      setPhoto(file);
      setFieldValue('photo', file);
    };
  };

  const defaultPreviewText = `Photo is optional. Max size is ${PRODUCT_PHOTO_MAX_SIZE_STRING}. Available extensions are JPG and PNG`;
  let preview = <SC.Preview>{defaultPreviewText}</SC.Preview>;
  let deleteBtn = null;
  let errorNode = null;

  if (error) {
    errorNode = <span className="error">{error}</span>;
  }

  if (photo || error) {
    preview = (
      <SC.Preview>
        <span className="file-data">Name: {photoName}</span>
        <span className="file-data">Size: {photoSize}</span>
      </SC.Preview>
    );

    deleteBtn = <Button clicked={resetState}>Delete</Button>;
  }

  return (
    <SC.Wrapper className="content">
      <div className="buttons">
        <label htmlFor="photo" className="label">
          <Button filled>Upload photo</Button>
        </label>
        {deleteBtn}
      </div>
      {preview}
      {errorNode}
      <input type="file" id="photo" name="photo" className="input" onChange={inputChangedHandler} />
    </SC.Wrapper>
  );
};

UploadPhoto.propTypes = {
  setFieldValue: PropTypes.func.isRequired,
};

export default UploadPhoto;
