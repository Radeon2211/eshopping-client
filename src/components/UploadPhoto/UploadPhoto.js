import { useRef, useState } from 'react';
import PropTypes from 'prop-types';
import * as SC from './UploadPhoto.sc';
import { isValidFileType, calculateFileSize } from '../../shared/utility/utility';
import Button from '../UI/Button/Button';
import FlexWrapper from '../UI/FlexWrapper';
import PlainText from '../UI/PlainText';
import theme from '../../styled/theme';
import { productPhotoFieldValues } from '../../shared/constants';

const PRODUCT_PHOTO_MAX_SIZE = 6291456;
const PRODUCT_PHOTO_MAX_SIZE_STRING = calculateFileSize(PRODUCT_PHOTO_MAX_SIZE);

export default function UploadPhoto({ setFieldValue, hasCurrentPhoto }) {
  const input = useRef(null);

  const [photo, setPhoto] = useState(null);
  const [photoName, setPhotoName] = useState('');
  const [photoSize, setPhotoSize] = useState('');
  const [error, setError] = useState('');
  const [isCurrentPhotoDeleted, setIsCurrentPhotoDeleted] = useState(!hasCurrentPhoto);

  const resetState = () => {
    setPhoto(null);
    setFieldValue('photo', null);
    setPhotoName('');
    setPhotoSize('');
    setError('');
    input.current.value = '';
  };

  const deleteCurrentPhotoHandle = () => {
    setIsCurrentPhotoDeleted(true);
    resetState();
    setFieldValue('photo', productPhotoFieldValues.DELETED);
  };

  const inputChangeHandle = (e) => {
    const { files } = e.target;
    if (!files.length > 0) {
      return;
    }

    const file = files[0];
    setPhotoName(file.name);
    setPhotoSize(calculateFileSize(file.size));

    if (!isValidFileType(file.type)) {
      setPhoto(null);
      setFieldValue('photo', productPhotoFieldValues.ERROR);
      setError('File extension is not valid (JPG and PNG only)');
      return;
    }

    if (file.size > PRODUCT_PHOTO_MAX_SIZE) {
      setPhoto(null);
      setFieldValue('photo', productPhotoFieldValues.ERROR);
      setError(`Maximum available size is ${PRODUCT_PHOTO_MAX_SIZE_STRING}`);
      return;
    }
    setError('');

    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onloadend = () => {
      setPhoto(file);
      setIsCurrentPhotoDeleted(true);
      setFieldValue('photo', file);
    };
  };

  const defaultPreviewText = `Photo is optional. Max size is ${PRODUCT_PHOTO_MAX_SIZE_STRING}. Available extensions are JPG and PNG`;
  let preview = (
    <PlainText $size="2" $mgTop="2" data-testid="UploadPhoto-default-preview">
      {defaultPreviewText}
    </PlainText>
  );
  let deleteThisBtn = null;
  let deleteCurrentBtn = null;
  let errorNode = null;

  if (error) {
    errorNode = (
      <PlainText $size="1" $mgTop="2" $color={theme.colors.red}>
        {error}
      </PlainText>
    );
  }

  if (photo || error) {
    preview = (
      <FlexWrapper $mgTop="2" $spacing="3" data-testid="UploadPhoto-preview-photo-info">
        <PlainText
          $size="1"
          $maxWidth="75%"
          $textOverflow="ellipsis"
          $whiteSpace="nowrap"
          $overflow="hidden"
        >
          Name: {photoName}
        </PlainText>
        <PlainText $size="1">Size: {photoSize}</PlainText>
      </FlexWrapper>
    );
    deleteThisBtn = (
      <Button $color="red" clicked={resetState}>
        Delete
      </Button>
    );
  }

  if (!isCurrentPhotoDeleted) {
    deleteCurrentBtn = (
      <Button $color="red" clicked={deleteCurrentPhotoHandle}>
        Delete current
      </Button>
    );
  }

  return (
    <SC.Wrapper>
      <FlexWrapper $spacing="3">
        <label htmlFor="photo" className="label">
          <Button $filled>Upload photo</Button>
        </label>
        {deleteThisBtn}
        {deleteCurrentBtn}
      </FlexWrapper>
      {preview}
      {errorNode}
      <input
        type="file"
        id="photo"
        name="photo"
        className="input"
        onChange={inputChangeHandle}
        data-testid="UploadPhoto-input"
        ref={input}
      />
    </SC.Wrapper>
  );
}

UploadPhoto.defaultProps = {
  hasCurrentPhoto: false,
};

UploadPhoto.propTypes = {
  setFieldValue: PropTypes.func.isRequired,
  hasCurrentPhoto: PropTypes.bool,
};
