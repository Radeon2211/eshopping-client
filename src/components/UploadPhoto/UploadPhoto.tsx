import { useRef, useState } from 'react';
import PropTypes from 'prop-types';
import * as SC from './UploadPhoto.sc';
import { isValidFileType, calculateFileSize } from '../../shared/utility/utility';
import Button from '../UI/Button/Button';
import FlexWrapper from '../UI/FlexWrapper';
import PlainText from '../UI/PlainText';
import theme from '../../styled/theme';
import { ProductPhotoFieldValue } from '../../shared/types/enums';

const PRODUCT_PHOTO_MAX_SIZE = 6291456;
const PRODUCT_PHOTO_MAX_SIZE_STRING = calculateFileSize(PRODUCT_PHOTO_MAX_SIZE);

interface UploadPhotoProps {
  setFieldValue: (fieldName: string, value: ProductPhotoFieldValue | File | null) => void;
  hasCurrentPhoto?: boolean;
}

export default function UploadPhoto({ setFieldValue, hasCurrentPhoto }: UploadPhotoProps) {
  const input = useRef<HTMLInputElement>(null);

  const [photo, setPhoto] = useState<File | null>(null);
  const [photoName, setPhotoName] = useState<string>('');
  const [photoSize, setPhotoSize] = useState<string>('');
  const [error, setError] = useState<string>('');
  const [isCurrentPhotoDeleted, setIsCurrentPhotoDeleted] = useState<boolean>(!hasCurrentPhoto);

  const resetState = () => {
    setPhoto(null);
    setFieldValue('photo', null);
    setPhotoName('');
    setPhotoSize('');
    setError('');
    if (input.current) {
      input.current.value = '';
    }
  };

  const deleteCurrentPhotoHandle = () => {
    setIsCurrentPhotoDeleted(true);
    resetState();
    setFieldValue('photo', ProductPhotoFieldValue.DELETED);
  };

  const inputChangeHandle = (e: React.ChangeEvent<HTMLInputElement>) => {
    const target = e.target as HTMLInputElement;
    if (!target.files || !target.files.length) {
      return;
    }

    const file = target.files[0];
    setPhotoName(file.name);
    setPhotoSize(calculateFileSize(file.size));

    if (!isValidFileType(file.type)) {
      setPhoto(null);
      setFieldValue('photo', ProductPhotoFieldValue.ERROR);
      setError('File extension is not valid (JPG and PNG only)');
      return;
    }

    if (file.size > PRODUCT_PHOTO_MAX_SIZE) {
      setPhoto(null);
      setFieldValue('photo', ProductPhotoFieldValue.ERROR);
      setError(`Maximum available size is ${PRODUCT_PHOTO_MAX_SIZE_STRING}`);
      return;
    }
    setError('');

    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onloadend = () => {
      setPhoto(file);
      setIsCurrentPhotoDeleted(true);
      setFieldValue('photo', file);
    };
  };

  const defaultPreviewText = `Photo is optional. Max size is ${PRODUCT_PHOTO_MAX_SIZE_STRING}. Available extensions are JPG and PNG`;
  let preview = (
    <PlainText $size="level2" $mgTop="level2" data-testid="UploadPhoto-default-preview">
      {defaultPreviewText}
    </PlainText>
  );
  let deleteThisBtn = null;
  let deleteCurrentBtn = null;
  let errorNode = null;

  if (error) {
    errorNode = (
      <PlainText $size="level1" $mgTop="level2" $color={theme.colors.red}>
        {error}
      </PlainText>
    );
  }

  if (photo || error) {
    preview = (
      <FlexWrapper $mgTop="level2" $spacing="level3" data-testid="UploadPhoto-preview-photo-info">
        <PlainText
          $size="level1"
          $maxWidth="75%"
          $textOverflow="ellipsis"
          $whiteSpace="nowrap"
          $overflow="hidden"
        >
          Name: {photoName}
        </PlainText>
        <PlainText $size="level1">Size: {photoSize}</PlainText>
      </FlexWrapper>
    );
    deleteThisBtn = (
      <Button $color="red" clicked={resetState}>
        Delete
      </Button>
    );
  }

  if (!isCurrentPhotoDeleted) {
    deleteCurrentBtn = (
      <Button $color="red" clicked={deleteCurrentPhotoHandle}>
        Delete current
      </Button>
    );
  }

  return (
    <SC.Wrapper>
      <FlexWrapper $spacing="level3">
        <label htmlFor="photo" className="label">
          <Button $filled>Upload photo</Button>
        </label>
        {deleteThisBtn}
        {deleteCurrentBtn}
      </FlexWrapper>
      {preview}
      {errorNode}
      <input
        type="file"
        id="photo"
        name="photo"
        className="input"
        onChange={inputChangeHandle}
        data-testid="UploadPhoto-input"
        ref={input}
      />
    </SC.Wrapper>
  );
}

UploadPhoto.defaultProps = {
  hasCurrentPhoto: false,
};

UploadPhoto.propTypes = {
  setFieldValue: PropTypes.func.isRequired,
  hasCurrentPhoto: PropTypes.bool,
};
