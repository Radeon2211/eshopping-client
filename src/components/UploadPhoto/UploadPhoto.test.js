import React from 'react';
import { render, cleanup, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeProvider } from 'styled-components';
import UploadPhoto from './UploadPhoto';
import theme from '../../styled/theme';
import { mockFile } from '../../shared/utility/utility';
import { productPhotoFieldValues } from '../../shared/constants';

const setUp = (hasCurrentPhoto, setFieldValue = jest.fn()) => {
  return render(
    <ThemeProvider theme={theme}>
      <UploadPhoto setFieldValue={setFieldValue} hasCurrentPhoto={hasCurrentPhoto} />
    </ThemeProvider>,
  );
};

afterEach(cleanup);

describe('<UploadPhoto />', () => {
  const maxPossibleFileSize = 6291456;
  const tooBigFileSize = 6400000;

  describe('check how renders', () => {
    it('should render default view when hasCurrentPhoto is false', () => {
      const { asFragment } = setUp(false);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render with delete current button when hasCurrentPhoto is true', () => {
      const { asFragment } = setUp(true);
      expect(asFragment()).toMatchSnapshot();
    });

    describe('successful upload', () => {
      it('should render delete button and info when photo is uploaded', async () => {
        const photo = mockFile.create('boots.jpg', maxPossibleFileSize, 'image/jpeg');
        const setFieldValueFn = jest.fn();

        const { asFragment } = setUp(false, setFieldValueFn);

        const uploadPhotoInput = screen.getByTestId('UploadPhoto-input');

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [photo] } });
        });
        await waitFor(() => {});

        expect(asFragment()).toMatchSnapshot();
        expect(setFieldValueFn).toHaveBeenCalledWith('photo', photo);
      });

      it('should render delete button and correct info when one photo is uploaded after another', async () => {
        const photo1 = mockFile.create('photo1.jpg', maxPossibleFileSize, 'image/jpeg');
        const photo2 = mockFile.create('photo2.jpg', maxPossibleFileSize, 'image/jpeg');
        const setFieldValueFn = jest.fn();

        const { asFragment } = setUp(false, setFieldValueFn);

        const uploadPhotoInput = screen.getByTestId('UploadPhoto-input');

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [photo1] } });
        });
        await waitFor(() => {});

        expect(setFieldValueFn).toHaveBeenNthCalledWith(1, 'photo', photo1);

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [photo2] } });
        });
        await waitFor(() => {});

        expect(asFragment()).toMatchSnapshot();
        expect(setFieldValueFn).toHaveBeenNthCalledWith(2, 'photo', photo2);
      });

      it('should render delete button and info when the same photo is uploaded twice', async () => {
        const photo = mockFile.create('boots.jpg', maxPossibleFileSize, 'image/jpeg');
        const setFieldValueFn = jest.fn();

        const { asFragment } = setUp(false, setFieldValueFn);

        const uploadPhotoInput = screen.getByTestId('UploadPhoto-input');

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [photo] } });
        });
        await waitFor(() => {});

        expect(setFieldValueFn).toHaveBeenNthCalledWith(1, 'photo', photo);

        await waitFor(() => {
          fireEvent.click(screen.getByText('Delete'));
        });
        await waitFor(() => {});

        expect(setFieldValueFn).toHaveBeenNthCalledWith(2, 'photo', null);

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [photo] } });
        });
        await waitFor(() => {});

        expect(asFragment()).toMatchSnapshot();
        expect(setFieldValueFn).toHaveBeenNthCalledWith(3, 'photo', photo);
      });
    });

    describe('show errors', () => {
      it('should render delete button and info when error appeared and then photo is uploaded correctly', async () => {
        const invalidPhoto = mockFile.create('boots.svg', maxPossibleFileSize, 'image/svg+xml');
        const validPhoto = mockFile.create('boots.png', maxPossibleFileSize, 'image/png');
        const setFieldValueFn = jest.fn();

        const { asFragment } = setUp(false, setFieldValueFn);

        const uploadPhotoInput = screen.getByTestId('UploadPhoto-input');

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [invalidPhoto] } });
        });
        await waitFor(() => {});

        expect(setFieldValueFn).toHaveBeenNthCalledWith(1, 'photo', productPhotoFieldValues.ERROR);
        expect(
          screen.getByText('File extension is not valid (JPG and PNG only)'),
        ).toBeInTheDocument();

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [validPhoto] } });
        });
        await waitFor(() => {});

        expect(asFragment()).toMatchSnapshot();
        expect(setFieldValueFn).toHaveBeenNthCalledWith(2, 'photo', validPhoto);
        expect(setFieldValueFn).toHaveBeenCalledTimes(2);
      });

      it('should render with error, delete button and file info when size is too big', async () => {
        const photo = mockFile.create('boots.png', tooBigFileSize, 'image/png');
        const setFieldValueFn = jest.fn();

        const { asFragment } = setUp(true, setFieldValueFn);

        const uploadPhotoInput = screen.getByTestId('UploadPhoto-input');

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [photo] } });
        });
        await waitFor(() => {});

        expect(asFragment()).toMatchSnapshot();
        expect(setFieldValueFn).toHaveBeenCalledWith('photo', productPhotoFieldValues.ERROR);
      });

      it('should render with error, delete button and file info when image type is incorrect', async () => {
        const photo = mockFile.create('boots.svg', maxPossibleFileSize, 'image/svg+xml');
        const setFieldValueFn = jest.fn();

        const { asFragment } = setUp(false, setFieldValueFn);

        const uploadPhotoInput = screen.getByTestId('UploadPhoto-input');

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [photo] } });
        });
        await waitFor(() => {});

        expect(asFragment()).toMatchSnapshot();
        expect(setFieldValueFn).toHaveBeenCalledWith('photo', productPhotoFieldValues.ERROR);
      });
    });

    describe('default view', () => {
      it('should render default view after deleting current photo', async () => {
        const setFieldValueFn = jest.fn();
        const { asFragment } = setUp(true, setFieldValueFn);

        await waitFor(() => {
          fireEvent.click(screen.getByText('Delete current'));
        });
        await waitFor(() => {});

        expect(asFragment()).toMatchSnapshot();
        expect(setFieldValueFn).toHaveBeenCalledWith('photo', productPhotoFieldValues.DELETED);
      });

      it('should render default view when files field has 0 length', async () => {
        const setFieldValueFn = jest.fn();
        const { asFragment } = setUp(false, setFieldValueFn);

        const uploadPhotoInput = screen.getByTestId('UploadPhoto-input');

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [] } });
        });
        await waitFor(() => {});

        expect(asFragment()).toMatchSnapshot();
        expect(setFieldValueFn).not.toHaveBeenCalled();
      });

      it('should render default view when file is uploaded and deleted', async () => {
        const photo = mockFile.create('boots.jpg', maxPossibleFileSize, 'image/jpeg');
        const setFieldValueFn = jest.fn();

        const { asFragment } = setUp(false, setFieldValueFn);

        const uploadPhotoInput = screen.getByTestId('UploadPhoto-input');

        await waitFor(() => {
          fireEvent.change(uploadPhotoInput, { target: { files: [photo] } });
        });
        await waitFor(() => {});

        expect(setFieldValueFn).toHaveBeenNthCalledWith(1, 'photo', photo);

        await waitFor(() => {
          fireEvent.click(screen.getByText('Delete'));
        });
        await waitFor(() => {});

        expect(asFragment()).toMatchSnapshot();
        expect(setFieldValueFn).toHaveBeenNthCalledWith(2, 'photo', null);
      });
    });
  });
});
