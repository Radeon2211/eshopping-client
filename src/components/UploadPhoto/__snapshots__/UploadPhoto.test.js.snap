// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadPhoto /> check how renders default view should render default view after deleting current photo 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hafmfB"
  >
    <div
      class="sc-gKsewC RnYls"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <label
        class="label"
        for="photo"
      >
        <button
          class="sc-gsTCUz gQZzmy"
          color="blue"
          type="button"
        >
          Upload photo
        </button>
      </label>
    </div>
    <span
      class="sc-eCssSg gXddCR"
      color=""
      display="inline-block"
      overflow="visible"
      size="2"
      spacing="0px"
      transform=""
    >
      Photo is optional. Max size is 6 MB. Available extensions are JPG and PNG
    </span>
    <input
      class="input"
      data-testid="UploadPhoto-input"
      id="photo"
      name="photo"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<UploadPhoto /> check how renders default view should render default view when files field has 0 length 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hafmfB"
  >
    <div
      class="sc-gKsewC RnYls"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <label
        class="label"
        for="photo"
      >
        <button
          class="sc-gsTCUz gQZzmy"
          color="blue"
          type="button"
        >
          Upload photo
        </button>
      </label>
    </div>
    <span
      class="sc-eCssSg gXddCR"
      color=""
      display="inline-block"
      overflow="visible"
      size="2"
      spacing="0px"
      transform=""
    >
      Photo is optional. Max size is 6 MB. Available extensions are JPG and PNG
    </span>
    <input
      class="input"
      data-testid="UploadPhoto-input"
      id="photo"
      name="photo"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<UploadPhoto /> check how renders should render default view when hasCurrentPhoto is false 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hafmfB"
  >
    <div
      class="sc-gKsewC RnYls"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <label
        class="label"
        for="photo"
      >
        <button
          class="sc-gsTCUz gQZzmy"
          color="blue"
          type="button"
        >
          Upload photo
        </button>
      </label>
    </div>
    <span
      class="sc-eCssSg gXddCR"
      color=""
      display="inline-block"
      overflow="visible"
      size="2"
      spacing="0px"
      transform=""
    >
      Photo is optional. Max size is 6 MB. Available extensions are JPG and PNG
    </span>
    <input
      class="input"
      data-testid="UploadPhoto-input"
      id="photo"
      name="photo"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<UploadPhoto /> check how renders should render with delete current button when hasCurrentPhoto is true 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hafmfB"
  >
    <div
      class="sc-gKsewC RnYls"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <label
        class="label"
        for="photo"
      >
        <button
          class="sc-gsTCUz gQZzmy"
          color="blue"
          type="button"
        >
          Upload photo
        </button>
      </label>
      <button
        class="sc-gsTCUz gqkUyy"
        color="red"
        type="button"
      >
        Delete current
      </button>
    </div>
    <span
      class="sc-eCssSg gXddCR"
      color=""
      display="inline-block"
      overflow="visible"
      size="2"
      spacing="0px"
      transform=""
    >
      Photo is optional. Max size is 6 MB. Available extensions are JPG and PNG
    </span>
    <input
      class="input"
      data-testid="UploadPhoto-input"
      id="photo"
      name="photo"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<UploadPhoto /> check how renders show errors should render delete button and info when error appeared and then photo is uploaded correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hafmfB"
  >
    <div
      class="sc-gKsewC RnYls"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <label
        class="label"
        for="photo"
      >
        <button
          class="sc-gsTCUz gQZzmy"
          color="blue"
          type="button"
        >
          Upload photo
        </button>
      </label>
      <button
        class="sc-gsTCUz gqkUyy"
        color="red"
        type="button"
      >
        Delete
      </button>
    </div>
    <div
      class="sc-gKsewC eJjdRD"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <span
        class="sc-eCssSg fwHiXJ"
        color=""
        display="inline-block"
        overflow="hidden"
        size="1"
        spacing="0px"
        transform=""
      >
        Name: boots.png
      </span>
      <span
        class="sc-eCssSg gTHaLR"
        color=""
        display="inline-block"
        overflow="visible"
        size="1"
        spacing="0px"
        transform=""
      >
        Size: 6 MB
      </span>
    </div>
    <input
      class="input"
      data-testid="UploadPhoto-input"
      id="photo"
      name="photo"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<UploadPhoto /> check how renders show errors should render with error when size is too big 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hafmfB"
  >
    <div
      class="sc-gKsewC RnYls"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <label
        class="label"
        for="photo"
      >
        <button
          class="sc-gsTCUz gQZzmy"
          color="blue"
          type="button"
        >
          Upload photo
        </button>
      </label>
      <button
        class="sc-gsTCUz gqkUyy"
        color="red"
        type="button"
      >
        Delete
      </button>
    </div>
    <div
      class="sc-gKsewC eJjdRD"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <span
        class="sc-eCssSg fwHiXJ"
        color=""
        display="inline-block"
        overflow="hidden"
        size="1"
        spacing="0px"
        transform=""
      >
        Name: boots.png
      </span>
      <span
        class="sc-eCssSg gTHaLR"
        color=""
        display="inline-block"
        overflow="visible"
        size="1"
        spacing="0px"
        transform=""
      >
        Size: 6,1 MB
      </span>
    </div>
    <span
      class="sc-eCssSg iNydNf"
      color="#F22613"
      display="inline-block"
      overflow="visible"
      size="1"
      spacing="0px"
      transform=""
    >
      Maximum available size is 6 MB
    </span>
    <input
      class="input"
      data-testid="UploadPhoto-input"
      id="photo"
      name="photo"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<UploadPhoto /> check how renders show errors should render with error, delete button and file info when image type is incorrect 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hafmfB"
  >
    <div
      class="sc-gKsewC RnYls"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <label
        class="label"
        for="photo"
      >
        <button
          class="sc-gsTCUz gQZzmy"
          color="blue"
          type="button"
        >
          Upload photo
        </button>
      </label>
      <button
        class="sc-gsTCUz gqkUyy"
        color="red"
        type="button"
      >
        Delete
      </button>
    </div>
    <div
      class="sc-gKsewC eJjdRD"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <span
        class="sc-eCssSg fwHiXJ"
        color=""
        display="inline-block"
        overflow="hidden"
        size="1"
        spacing="0px"
        transform=""
      >
        Name: boots.svg
      </span>
      <span
        class="sc-eCssSg gTHaLR"
        color=""
        display="inline-block"
        overflow="visible"
        size="1"
        spacing="0px"
        transform=""
      >
        Size: 6 MB
      </span>
    </div>
    <span
      class="sc-eCssSg iNydNf"
      color="#F22613"
      display="inline-block"
      overflow="visible"
      size="1"
      spacing="0px"
      transform=""
    >
      File extension is not valid (JPG and PNG only)
    </span>
    <input
      class="input"
      data-testid="UploadPhoto-input"
      id="photo"
      name="photo"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<UploadPhoto /> check how renders successful upload should render delete button and info when photo is uploaded 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hafmfB"
  >
    <div
      class="sc-gKsewC RnYls"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <label
        class="label"
        for="photo"
      >
        <button
          class="sc-gsTCUz gQZzmy"
          color="blue"
          type="button"
        >
          Upload photo
        </button>
      </label>
      <button
        class="sc-gsTCUz gqkUyy"
        color="red"
        type="button"
      >
        Delete
      </button>
    </div>
    <div
      class="sc-gKsewC eJjdRD"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <span
        class="sc-eCssSg fwHiXJ"
        color=""
        display="inline-block"
        overflow="hidden"
        size="1"
        spacing="0px"
        transform=""
      >
        Name: boots.jpg
      </span>
      <span
        class="sc-eCssSg gTHaLR"
        color=""
        display="inline-block"
        overflow="visible"
        size="1"
        spacing="0px"
        transform=""
      >
        Size: 6 MB
      </span>
    </div>
    <input
      class="input"
      data-testid="UploadPhoto-input"
      id="photo"
      name="photo"
      type="file"
    />
  </div>
</DocumentFragment>
`;
